tests/group_functions/group_scan.h

```diff
@@ -303,6 +303,10 @@ void check_scan_over_group(sycl::queue& queue, sycl::range<D> range, OpT op,
   bool ret_type_e = false;
   bool ret_type_i = false;
 
+  if (std::sqrt(std::numeric_limits<T>::max()) + T(init) <= range.size()) {
+    return;
+  }
+
```

Comment:
I don't think it's a good idea silently skipping all checks in this case.
Although this sounds like a test issue, rather than a bug in the tested implementation, we probably can use REQUIRE macro to "Check that the maximum value of T is large enough to store the results of the scan functions.".

Reply:
Why do we launch the configuration causing the overflow? Isn't this a bug?

Reply:
Right. This where I think we should change the logic to avoid an overflow. 
work_goup_size here is the maximum possible # of work-items the device can execute. Do we really need to the maximum # of work-items to test the scan functionality? If so, the input buffer must be initialized with zeros to avoid overflow.
But I don't think we need to run so many work-items, it's a good stress test, but an overkill for this test purpose.

https://github.com/KhronosGroup/SYCL-CTS/blob/47a05179ba09a364c4db03ce07fc387564eebdac/tests/group_functions/group_scan.h#L68 - this seems to be the right place to make sure that we can run the test w/o an overflow. BTW, it's better to use [the sum of arithmetic progression](https://en.wikipedia.org/wiki/Arithmetic_progression#Sum) instead of sqrt for the overflow check.